components/constellation/constellation.rs

```diff
@@ -1722,7 +1760,7 @@ impl<Message, LTF, STF> Constellation<Message, LTF, STF>
             TraversalDirection::Forward(delta) => {
                 for entry in self.joint_session_future(top_level_frame_id).take(delta) {
```

Comment:
Could we use something like `self.joint_session_future(top_level_frame_id).take(delta).collect()` here?